libsignal-service/src/gv2/api.rs

```diff
@@ -0,0 +1,195 @@
+use std::{
+    collections::HashMap,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+use crate::{
+    configuration::Endpoint,
+    prelude::{PushService, ServiceError},
+    proto::DecryptedGroup,
+    push_service::HttpCredentials,
+};
+
+use rand::RngCore;
+use uuid::Uuid;
+use zkgroup::{
+    auth::AuthCredentialResponse, groups::GroupSecretParams, ServerPublicParams,
+};
+
+use super::{models::CredentialResponse, operations::GroupOperations};
+
+#[derive(Debug, thiserror::Error)]
+pub enum CredentialsCacheError {
+    #[error("failed to read values from cache: {0}")]
+    ReadError(String),
+    #[error("failed to write values from cache: {0}")]
+    WriteError(String),
+}
+
+pub trait CredentialsCache {
```

Comment:
Can do with some documentation now, maybe some example code in the doc string.

Reply:
since now we have `InMemoryCredentialsCache` it shouldn't be a problem, I added a comment though.